envoy/api/v2/route/route.proto

```diff
@@ -803,7 +814,36 @@ message HeaderMatcher {
   // * The regex *\d{3}* matches the value *123*
   // * The regex *\d{3}* does not match the value *1234*
   // * The regex *\d{3}* does not match the value *123.456*
-  google.protobuf.BoolValue regex = 3;
+  google.protobuf.BoolValue regex = 3 [deprecated = true];
+
+  // [#not-implemented-hide:]
+  // Specifies how the header match will be performed to route the request.
+  // If header_match_specifier is absent, a request that has the name header will match, regardless
+  // of the header's value.
+  oneof header_match_specifier {
+    // [#not-implemented-hide:]
+    // If specified, header match will be performed based on the value of the header.
+    string value_match = 4;
+
+    // [#not-implemented-hide:]
```

Comment:
If you have an outer scope not-implemented-hide, it applies to the entire subtree in AST.

Reply:
ok, will remove these tags.

Reply:
@htuch  looks like that doesn't work for oneofs. I got a warning/error while building documentation:
 envoy_api_msg_route.int64range (if the link has no caption the label must precede a section header)